File: src/main/java/com/itemlog/model/LocationData.java
package com.itemlog.model;

import org.jetbrains.annotations.Nullable;

public record LocationData(
    @Nullable String world,
    double x, double y, double z,
    float yaw, float pitch
) {
    public static LocationData from(org.bukkit.Location loc) {
        return new LocationData(
            loc.getWorld() != null ? loc.getWorld().getName() : null,
            loc.getX(), loc.getY(), loc.getZ(),
            loc.getYaw(), loc.getPitch()
        );
    }
}

File: src/main/kotlin/com/itemlog/model/LocationData.kt
package com.itemlog.model

import org.bukkit.Location
import org.jetbrains.annotations.Nullable

data class LocationData(
    @Nullable val world: String?,
    val x: Double,
    val y: Double,
    val z: Double,
    val yaw: Float,
    val pitch: Float,
) {
    companion object {
        @Nullable
        fun from(loc: Location?): LocationData? {
            if (loc == null) return null
            return LocationData(
                loc.world?.name,
                loc.x,
                loc.y,
                loc.z,
                loc.yaw,
                loc.pitch,
            )
        }
    }
}
